service/telemetry.go

```diff
@@ -132,21 +131,21 @@ func (tel *colTelemetry) initOnce(asyncErrorChannel chan<- error, ballastSizeByt
 	go func() {
 		serveErr := tel.server.ListenAndServe()
 		if serveErr != nil && serveErr != http.ErrServerClosed {
-			asyncErrorChannel <- serveErr
+			col.asyncErrorChannel <- serveErr
 		}
 	}()
 
 	return nil
 }
 
-func (tel *colTelemetry) initOpenCensus(level configtelemetry.Level, instanceID string, ballastSizeBytes uint64) (http.Handler, error) {
-	processMetricsViews, err := telemetry2.NewProcessMetricsViews(ballastSizeBytes)
+func (tel *colTelemetry) initOpenCensus(col *Collector, instanceID string) (http.Handler, error) {
```

Comment:
It feels very unnatural to pass Collector instance (as argument) when the Collector instance owns the telemetry. I think this is not a clean design.

Also this makes thinks very unclear about concurrency, if this func changes anything in the collector.

Reply:
I did this because the parameter list was growing significantly and all of the required information (with the exception of `instanceID`) is available from the `Collector` struct.  I don' think that methods with ever-expanding parameter lists is good design, either.